src/table.rs

```diff
@@ -0,0 +1,2013 @@
+        }
+    }
+
+    /// Returns an `Entry` for an entry in the table with the given hash
+    /// and which satisfies the equality function passed.
+    ///
+    /// This can be used to remove the entry from the table, or insert a new
+    /// entry with the given hash if one doesn't already exist.
+    ///
+    /// This method will call `eq` for all entries with the given hash, but may
+    /// also call it for entries with a different hash. `eq` should only return
+    /// true for the desired entry, at which point the search is stopped.
+    ///
+    /// This method may grow the table in preparation for an insertion. Call
+    /// [`HashTable::find_entry`] if this is undesirable.
+    ///
+    /// `hasher` is called if entries need to be moved or copied to a new table.
+    /// This must return the same hash value that each entry was inserted with.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(feature = "nightly")]
+    /// # fn test() {
+    /// use ahash::AHasher;
+    /// use hashbrown::hash_table::Entry;
+    /// use hashbrown::HashTable;
+    /// use std::hash::{BuildHasher, BuildHasherDefault};
+    ///
+    /// let mut table = HashTable::new();
+    /// let hasher = BuildHasherDefault::<AHasher>::default();
+    /// let hasher = |val: &_| hasher.hash_one(val);
+    /// table.insert_unchecked(hasher(&1), (1, "a"), |val| hasher(&val.0));
+    /// if let Entry::Occupied(entry) = table.entry(hasher(&1), |val| val.0 == 1, |val| hasher(&val.0))
+    /// {
+    ///     entry.remove();
+    /// }
+    /// if let Entry::Vacant(entry) = table.entry(hasher(&2), |val| val.0 == 2, |val| hasher(&val.0)) {
+    ///     entry.insert((2, "b"));
+    /// }
+    /// assert_eq!(table.find(hasher(&1), |val| val.0 == 1), None);
+    /// assert_eq!(table.find(hasher(&2), |val| val.0 == 2), Some(&(2, "b")));
+    /// # }
+    /// # fn main() {
+    /// #     #[cfg(feature = "nightly")]
+    /// #     test()
+    /// # }
+    /// ```
+    pub fn entry(
+        &mut self,
+        hash: u64,
+        eq: impl FnMut(&T) -> bool,
+        hasher: impl Fn(&T) -> u64,
+    ) -> Entry<'_, T, A> {
+        match self.raw.find_or_find_insert_slot(hash, eq, hasher) {
+            Ok(bucket) => Entry::Occupied(OccupiedEntry {
+                hash,
+                bucket,
+                table: self,
+            }),
+            Err(insert_slot) => Entry::Vacant(VacantEntry {
+                hash,
+                insert_slot,
+                table: self,
+            }),
+        }
+    }
+
+    /// Inserts an element into the `HashTable` with the given hash value, but
+    /// without checking whether an equivalent element already exists within the
+    /// table.
+    ///
+    /// This is
+    ///
+    /// `hasher` is called if entries need to be moved or copied to a new table.
+    /// This must return the same hash value that each entry was inserted with.
+    pub fn insert_unchecked(
```

Comment:
I do prefer the `insert_unique` naming as the `unchecked` postfix makes me think it's memory unsafe.